Specs/Core/sampleTerrainMostDetailedSpec.js

```diff
@@ -0,0 +1,93 @@
+/*global defineSuite*/
+defineSuite([
+        'Core/sampleTerrainMostDetailed',
+        'Core/Cartographic',
+        'Core/CesiumTerrainProvider',
+        'ThirdParty/when'
+    ], function(
+        sampleTerrainMostDetailed,
+        Cartographic,
+        CesiumTerrainProvider,
+        when) {
+    "use strict";
+
+    var terrainProvider = new CesiumTerrainProvider({
+        url : '//assets.agi.com/stk-terrain/world'
+    });
+
+    it('queries heights', function() {
+        var positions = [
+                         Cartographic.fromDegrees(86.925145, 27.988257),
+                         Cartographic.fromDegrees(87.0, 28.0)
+                     ];
+
+        return sampleTerrainMostDetailed(terrainProvider, positions).then(function(passedPositions) {
+            expect(passedPositions).toBe(positions);
+            expect(positions[0].height).toBeGreaterThan(5000);
+            expect(positions[0].height).toBeLessThan(10000);
+            expect(positions[1].height).toBeGreaterThan(5000);
+            expect(positions[1].height).toBeLessThan(10000);
+        });
+    });
+
+    it('throw querying heights from Small Terrain', function() {
+        var terrainProvider = new CesiumTerrainProvider({
+            url : '//cesiumjs.org/smallTerrain'
+        });
+
+        var positions = [
+                         Cartographic.fromDegrees(86.925145, 27.988257),
+                         Cartographic.fromDegrees(87.0, 28.0)
+                     ];
+
+        expect(function() {
+            sampleTerrainMostDetailed(terrainProvider, positions);
```

Comment:
Not sure of the correct syntax to catch the exception being thrown by CesiumTerrainProvider.getMaximumTileLevel().